backend/prompt_studio/prompt_studio_index_manager/models.py

```diff
@@ -78,3 +88,28 @@ class Meta:
                 name="unique_document_manager_profile_manager",
             ),
         ]
+
+
+# Function will be executed every time an instance of IndexManager is deleted.
+@receiver(post_delete, sender=IndexManager)
+def perform_vector_db_cleanup(sender, instance, **kwargs):
+    """Signal to perform vector db cleanup."""
+    logger.info("Performing vector db cleanup")
+    try:
+        # Get the index_ids_history to clean up from the vector db
+        index_ids_history = json.loads(instance.index_ids_history)
+        vector_db_instance_id = str(instance.profile_manager.vector_store.id)
```

Comment:
As discussed, please be aware that the vector DB used during indexing might be changed / removed when `IndexManager` is deleted

Reply:
This is to be discussed further and taken up as a separate task.